code/modules/unit_tests/say.dm

```diff
@@ -40,3 +40,92 @@
 // For the above test to track the last use of say's message args.
 /mob/living
 	var/last_say_args_ref
+
+/datum/unit_test/translate_language
+	var/mob/host_mob
+
+/datum/unit_test/translate_language/Run()
+	host_mob = allocate(/mob/living/carbon/human)
+	var/surfer_quote = "surfing in the USA"
+
+	host_mob.grant_language(/datum/language/beachbum, spoken=TRUE, understood=FALSE) // can speak but can't understand
+	host_mob.add_blocked_language(subtypesof(/datum/language) - /datum/language/beachbum, LANGUAGE_STONER)
+	TEST_ASSERT_NOTEQUAL(surfer_quote, host_mob.translate_language(host_mob, /datum/language/beachbum, surfer_quote), "Language test failed. Mob was supposed to understand: [surfer_quote]")
+
+	host_mob.grant_language(/datum/language/beachbum, spoken=TRUE, understood=TRUE) // can now understand
+	TEST_ASSERT_EQUAL(surfer_quote, host_mob.translate_language(host_mob, /datum/language/beachbum, surfer_quote), "Language test failed. Mob was supposed NOT to understand: [surfer_quote]")
+
+/datum/unit_test/speech
```

Comment:
Document the test

Reply:
Done with https://github.com/tgstation/tgstation/pull/69799/commits/c2424862585f3e2ab030c2d8e7c749959c2cc4de